DSharpPlus/Clients/DiscordClient.Dispatch.cs

```diff
@@ -36,6 +36,7 @@ private async Task ReceiveGatewayEventsAsync()
     {
         while (!this.eventReader.Completion.IsCompleted)
         {
+            // Open question: Do we want to pass a CT here?

```

Comment:
Yes because this loop waits until the next recieved gw event

Reply:
The CT is cancelled further up, which should in theory gracefully close the channel, instead of outright aborting it? ChannelClosedException vs OperationCancelledException, I'll have to look at this more closely.

Reply:
have you perchance looked at this more closely?

Reply:
I can when I'm not 6 hours past my sleep schedule

Reply:
It's valid to be thorough and add an extra check if we're not sure. Or- like aki said, double check :))